src/test/java/com/xceptance/neodymium/util/SelenideAddonsTest.java

```diff
@@ -262,6 +262,120 @@ public void testSafeRunnable()
+        testSafeRunnable(runArray);
+    }
+
+    @Test()
+    public void testSafeSupplierInCause()
+    {
+        List<Runnable> runArray = new ArrayList<Runnable>();
+        runArray.add(
+                     () -> {
+                         throw UIAssertionError.wrap(WebDriverRunner.driver(),
+                                                     new AssertionError(" value displayed:StaleElementReferenceException: stale element reference: element is not attached to the page document>StaleElementReferenceException: stale element reference: element is not attached to the page document</StaleElementReferenceException: stale element reference: element is not attached to the page document>'\n"
+                                                                        +
+                                                                        "Actual value: StaleElementReferenceException: stale element refe"),
+                                                     0);
+                     });
+        runArray.add(
+                     () -> {
+                         throw UIAssertionError.wrap(WebDriverRunner.driver(),
+                                                     new AssertionError(" value displayed:StaleElementReferenceException: stale element reference: element is not attached to the page document>StaleElementReferenceException: stale element reference: element is not attached to the page document</StaleElementReferenceException: stale element reference: element is not attached to the page document>'\n"
+                                                                        +
+                                                                        "Actual value: StaleElementReferenceException: stale element refe"),
+                                                     0);
+                     });
+        runArray.add(
+                     () -> {
+                         throw UIAssertionError.wrap(WebDriverRunner.driver(),
+                                                     new AssertionError(" value displayed:StaleElementReferenceException: stale element reference: element is not attached to the page document>StaleElementReferenceException: stale element reference: element is not attached to the page document</StaleElementReferenceException: stale element reference: element is not attached to the page document>'\n"
+                                                                        +
+                                                                        "Actual value: StaleElementReferenceException: stale element refe"),
+                                                     0);
+                     });
+        runArray.add(
+                     () -> {
+                         throw new StaleElementReferenceException("You shall pass!");
+                     });
+        runArray.add(
+                     () -> {
+                         throw new StaleElementReferenceException("You shall never be seen!");
```

Comment:
Please change this also to be an UIAssertionError

Reply:
I think you need to an additional version of the testSafeSupplier function to cover this. Otherwise we can not be completely sure that we cover everything.

Reply:
Maybe we can refacture the two private utility functions to once since they show a lot of duplicate code.

Reply:
Two private methods are now merged to one and the code is changed to be more readable